server/rest/image.py

```diff
@@ -315,8 +311,20 @@ def updateImageStatus(self, event):
                 job['args'][0], force=True)
             if status == JobStatus.SUCCESS:
                 result = getCeleryApp().AsyncResult(job['celeryTaskId']).get()
-                image['digest'] = result['Id']
-                image['status'] = ImageStatus.AVAILABLE
+                digest = result['image_digest']
+                if digest:
+                    if '@' in digest:
+                        # Grab just the SHA from the end of the digest
+                        image['digest'] = digest.split('@')[1]
+                        image['status'] = ImageStatus.AVAILABLE
+                    else:
+                        image['status'] = ImageStatus.INVALID
+                        raise RestException(
+                            'Invalid image digest produced for ' + str(image['_id']) + ': ' + digest
+                        )
+                else:
+                    image['status'] = ImageStatus.INVALID
+                    raise RestException('No image digest produced for ' + str(image['_id']))
```

Comment:
There are a couple of issues with raising exception here:
1. It's not resource, but an event handler. This exception will never get propagated to UI from here.
2. It will prevent image model from being updated with `INVALID` status, cause it will never reach L324

I'd suggest logging error instead.

Reply:
Good call, I had not thought about the asynchronicity here.